### PR TITLE
[gardening] Remove unused diagnostics. Fix end of namespace comments. Match parameter name. Use override as appropriate.

### PR DESCRIPTION
**Pull request by practicalswift**

### CODE REVIEW

**Comment by practicalswift** [comment]
@swift-ci please test

**Comment by practicalswift** [comment]
@swift-ci please smoke test OS X platform

**Comment by practicalswift** [comment]
@swift-ci please test OS X platform

File: lib/AST/ASTPrinter.cpp [hunk 1]
```diff
@@ -1266,7 +1266,7 @@ struct RequirementPrintLocation {
   /// Whether the requirement needs to be in a where clause.
   bool InWhereClause;
 };
-};
+} // namespace
```
Comment (jrose-apple) [inline lib/AST/ASTPrinter.cpp]: Think you missed the anonymous namespace special case again.

Comment (practicalswift) [inline lib/AST/ASTPrinter.cpp]: Fixed in #8893 :-)

**Pull request merged**